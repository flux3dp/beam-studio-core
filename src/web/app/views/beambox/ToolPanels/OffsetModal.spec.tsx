/* eslint-disable import/first */
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const onCancel = jest.fn();
const onOk = jest.fn();

const get = jest.fn();
jest.mock('implementations/storage', () => ({
  get,
}));

import OffsetModal from './OffsetModal';

describe('should render correctly', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  test('default unit is mm', () => {
    get.mockReturnValue(undefined);
    const wrapper = shallow(<OffsetModal onCancel={onCancel} onOk={onOk} />);
    expect(get).toBeCalledTimes(1);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('ForwardRef(Select)').at(0).props().onChange(0);
    expect(wrapper.find('ForwardRef(Select)').at(0).props().value).toBe(0);
    wrapper.find('ForwardRef(Select)').at(1).props().onChange('round');
    expect(wrapper.find('ForwardRef(Select)').at(1).props().value).toBe('round');
    wrapper.find('InputNumber').at(0).props().onChange(10);
    expect(wrapper.find('Slider').at(0).props().value).toBe(10);
    expect(wrapper.find('InputNumber').at(0).props().value).toBe(10);
    wrapper.find('Modal').props().onOk();
    expect(onOk).toBeCalledTimes(1);
    expect(onOk).toBeCalledWith({ dir: 0, distance: 10, cornerType: 'round' });
  });

  test('default unit is inches', () => {
    get.mockReturnValue('inches');
    const wrapper = shallow(<OffsetModal onCancel={onCancel} onOk={onOk} />);
    expect(get).toBeCalledTimes(1);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('Slider').at(0).props().onChange(1.5);
    expect(wrapper.find('Slider').at(0).props().value).toBe(1.5);
    expect(wrapper.find('InputNumber').at(0).props().value).toBe(1.5);
    wrapper.find('Modal').props().onOk();
    expect(onOk).toBeCalledTimes(1);
    expect(onOk).toBeCalledWith({ dir: 1, distance: 38.1, cornerType: 'sharp' });
  });
});
