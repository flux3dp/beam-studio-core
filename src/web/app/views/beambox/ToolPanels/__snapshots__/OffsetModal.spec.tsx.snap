// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly default unit is inches 1`] = `
<ConfigProvider
  theme={
    {
      "components": {
        "Button": {
          "borderRadius": 100,
        },
        "InputNumber": {
          "borderRadius": 100,
        },
        "Select": {
          "borderRadius": 100,
        },
      },
    }
  }
>
  <Modal
    cancelText="Cancel"
    centered={true}
    className="modal"
    closeIcon={
      <AntdIcon
        className="close-icon"
        component="svgr-url"
        viewBox="0 0 32 32"
      />
    }
    okText="Confirm"
    onCancel={[MockFunction]}
    onOk={[Function]}
    open={true}
  >
    <div
      className="title"
    >
      Offset
    </div>
    <div
      className="field"
    >
      <span
        className="label"
      >
        Offset Direction
      </span>
      <ForwardRef(Select)
        className="select"
        dropdownMatchSelectWidth={false}
        onChange={[Function]}
        options={
          [
            {
              "label": "Outward",
              "value": 1,
            },
            {
              "label": "Inward",
              "value": 0,
            },
          ]
        }
        value={1}
      />
    </div>
    <div
      className="field"
    >
      <span
        className="label"
      >
        Corner
      </span>
      <ForwardRef(Select)
        className="select"
        dropdownMatchSelectWidth={false}
        onChange={[Function]}
        options={
          [
            {
              "label": "Sharp",
              "value": "sharp",
            },
            {
              "label": "Round",
              "value": "round",
            },
          ]
        }
        value="sharp"
      />
    </div>
    <div
      className="field"
    >
      <span
        className="label"
      >
        Offset Distance
      </span>
      <InputNumber
        className="input with-unit"
        controls={false}
        min={0}
        onChange={[Function]}
        precision={2}
        prefix={
          <span
            className="unit"
          >
            inch
          </span>
        }
        type="number"
        value={0.2}
      />
      <Slider
        className="slider"
        max={1}
        min={0}
        onChange={[Function]}
        step={0.1}
        value={0.2}
      />
    </div>
  </Modal>
</ConfigProvider>
`;

exports[`should render correctly default unit is mm 1`] = `
<ConfigProvider
  theme={
    {
      "components": {
        "Button": {
          "borderRadius": 100,
        },
        "InputNumber": {
          "borderRadius": 100,
        },
        "Select": {
          "borderRadius": 100,
        },
      },
    }
  }
>
  <Modal
    cancelText="Cancel"
    centered={true}
    className="modal"
    closeIcon={
      <AntdIcon
        className="close-icon"
        component="svgr-url"
        viewBox="0 0 32 32"
      />
    }
    okText="Confirm"
    onCancel={[MockFunction]}
    onOk={[Function]}
    open={true}
  >
    <div
      className="title"
    >
      Offset
    </div>
    <div
      className="field"
    >
      <span
        className="label"
      >
        Offset Direction
      </span>
      <ForwardRef(Select)
        className="select"
        dropdownMatchSelectWidth={false}
        onChange={[Function]}
        options={
          [
            {
              "label": "Outward",
              "value": 1,
            },
            {
              "label": "Inward",
              "value": 0,
            },
          ]
        }
        value={1}
      />
    </div>
    <div
      className="field"
    >
      <span
        className="label"
      >
        Corner
      </span>
      <ForwardRef(Select)
        className="select"
        dropdownMatchSelectWidth={false}
        onChange={[Function]}
        options={
          [
            {
              "label": "Sharp",
              "value": "sharp",
            },
            {
              "label": "Round",
              "value": "round",
            },
          ]
        }
        value="sharp"
      />
    </div>
    <div
      className="field"
    >
      <span
        className="label"
      >
        Offset Distance
      </span>
      <InputNumber
        className="input with-unit"
        controls={false}
        min={0}
        onChange={[Function]}
        precision={2}
        prefix={
          <span
            className="unit"
          >
            mm
          </span>
        }
        type="number"
        value={5}
      />
      <Slider
        className="slider"
        max={20}
        min={1}
        onChange={[Function]}
        value={5}
      />
    </div>
  </Modal>
</ConfigProvider>
`;
